static/js/components/ShiftPage.jsx

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Typography from "@material-ui/core/Typography";
+import Paper from "@material-ui/core/Paper";
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
+import PropTypes from "prop-types";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import NewShift from "./NewShift";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    maxWidth: "22.5rem",
+    backgroundColor: theme.palette.background.paper,
+  },
+  paperContent: {
+    padding: "1rem",
+  },
+}));
+
+const textStyles = makeStyles(() => ({
+  primary: {
+    fontWeight: "bold",
+    fontSize: "110%",
+  },
+}));
+
+export function shiftTitle(shift) {
+  if (!shift?.group.name) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const result = `${shift?.group.name}`;
+  return result;
+}
+
+export function shiftInfo(shift) {
+  if (!shift?.group.name) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const array = [
+    ...(shift?.start_date ? [`Start Date: ${shift.start_date}`] : []),
+    ...(shift?.end_date ? [`End Date: ${shift.end_date}`] : []),
+  ];
+
+  // eslint-disable-next-line prefer-template
+  const result = "( " + array.join(" / ") + " )";
+
+  return result;
+}
+
+const ShiftList = ({ shifts }) => {
+  const classes = useStyles();
+  const textClasses = textStyles();
+
+  return (
+    <div className={classes.root}>
+      <List component="nav">
+        {shifts?.map((shift) => (
+          <ListItem button key={shift.id}>
+            <ListItemText
+              primary={shiftTitle(shift)}
+              secondary={shiftInfo(shift)}
+              classes={textClasses}
+            />
+          </ListItem>
+        ))}
+      </List>
+    </div>
+  );
+};
+
+const ShiftPage = () => {
+  const classes = useStyles();
+  const { shiftList } = useSelector((state) => state.shifts);
+
+  if (!shiftList) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <Grid container spacing={3}>
+      <Grid item md={6} sm={12}>
+        <Paper elevation={1}>
+          <div className={classes.paperContent}>
+            <Typography variant="h6">List of Shifts</Typography>
+            <ShiftList shifts={shiftList} />
+          </div>
+        </Paper>
+      </Grid>
+      <Grid item md={6} sm={12}>
+        <Paper>
+          <div className={classes.paperContent}>
+            <Typography variant="h6">Add a New Shift</Typography>
+            <NewShift />
```

Comment:
Only show this if the user has the appropriate role

Reply:
We want users to be able to add normal shifts right?

Reply:
I guess this can be dictated by "Manage shifts" going forward :)